### PR TITLE
Graphs API

### PR DESCRIPTION
**Pull request by Nirmitjatana**
### Highlight points
- Added functions to get response time data for day, week, month and year.
- Normalized data with average of available response times.
- Handled edge cases for number of data points.

### How to use:
Call specific functions `Dayvalues , Weekvalues, Monthvalues, Yearvalues` with name of the website as slug to fetch data points to plot graph.

### CODE REVIEW

**Comment by Nirmitjatana** [comment]
closes #36

File: info.log [hunk 1]
```diff
@@ -0,0 +1,594 @@
+2021-09-20T18:35:15.515Z info: [main df2104d] ⭕ Create incidents.yml
```
Comment (jayantkatia) [inline info.log:1]: *.log files shouldn't be committed. May delete *.log files in an upcoming commit.

File: src/commands/run.ts [hunk 2]
```diff
@@ -41,11 +44,17 @@ async run() {
   // It would be desirable to execute each iteration of CI in a cycle
   if (flags.iterations) {
     for (let i = 0; i < flags.iterations; i++) {
+      console.log("running for "+i+1+"th iteration")
       if (flags.uptime)
         queue.add(update)
       if (flags.responseTime)
         queue.add(() => update(true))
       if (flags.graphs)
+        // method to call data point function
```
Comment (jayantkatia) [inline src/commands/run.ts:53]: Hard-coded values

File: src/commands/run.ts [hunk 3]
```diff
@@ -41,11 +44,17 @@ async run() {
   // It would be desirable to execute each iteration of CI in a cycle
   if (flags.iterations) {
     for (let i = 0; i < flags.iterations; i++) {
+      console.log("running for "+i+1+"th iteration")
```
Comment (jayantkatia) [inline src/commands/run.ts:47]: Use the non-blocking `this.log()` instead.

File: src/graphs.ts [hunk 4]
```diff
@@ -1,120 +1,207 @@
...
-      grid: {
-        display: false,
-      },
-    },
-  },
+// Error.stackTraceLimit = Infinity;
+export const Dayvalues = async(slug = "") =>{
+    try{
+        const config = await getConfig()
+        const exists = config.sites.map(ob => ob.name === `${slug}` ?  true : false)
```
Comment (jayantkatia) [inline src/graphs.ts:11]: Double check. ```ob.name``` may not be the slug (Did you mean ```ob.slug || slugify(ob.name)```?). Correct semantics, if meant site-name. But best way is to use slugs.

File: src/helpers/graphs.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    const tYear = responseTimes.filter(i => dayjs(i[0]).isAfter(dayjs().subtract(1, 'year')))
+    const dataItems: [string, [string, number][]][] = [
+      [`${slug}/response-time-day.yml`, tDay],
+      [`${slug}/response-time-week.yml`, tWeek],
+      [`${slug}/response-time-month.yml`, tMonth],
+      [`${slug}/response-time-year.yml`, tYear],
+    ]
+
+    for await (const dataItem of dataItems) {
+      await ensureFile(join('.', 'history', 'response-data' ,dataItem[0]))
```
Comment (jayantkatia) [inline src/helpers/graphs.ts:60]: dataItem[0] is equal to ```${slug}/response-time-day.yml```. Confirm whether it works with Windows OS that uses ```\```, If it doesn't break it down to, ```join('.', 'history', 'response-data' ,slug, dataItem[0].split('/')[1] )```

File: src/commands/run.ts (hunk 2 above)
Comment (Nirmitjatana) [reply src/commands/run.ts:53]: removed them, this was just for reference.

File: src/helpers/graphs.ts (hunk 5 above)
Comment (Nirmitjatana) [reply src/helpers/graphs.ts:60]: This works for windows. Anyways I have changed it for best practice.

**Comment by Nirmitjatana** [comment]
@jayantkatia can you check this out.

**Review by jayantkatia — APPROVED** [review]

**Comment by jayantkatia** [comment]
LGTM :rocket:

**Pull request merged**